ext/stringio/stringio.c

```diff
@@ -130,6 +131,16 @@ strio_substr(struct StringIO *ptr, long pos, long len, rb_encoding *enc)
 
 #define StringIO(obj) get_strio(obj)
 
+// From internal/string.h
+// Ruby 3.4 introduces "chilled" strings, which behave like frozen strings unless written to.
+// In order to facilitate an easier upgrade path, print a deprecation warning instead of raising.
+#define STR_CHILLED FL_USER3
```

Comment:
Yeah, this is a big no no. 

I'll see if I can find a better solution.

Reply:
Thank you for looking into it

Reply:
Here's my solution: https://github.com/ruby/stringio/pull/93

Reply:
That's certainly better